TrackletAlgorithm/VMRouter.h

```diff
@@ -657,27 +657,33 @@ void VMRouter(const BXType bx, BXType& bx_o, const int fineBinTable[], const int
 	ap_uint<kNBits_MemAddr> addrCountOL[nvmOL][MaxOLCopies]; // Writing of TE Overlap stubs
 	ap_uint<kNBits_MemAddr> addrCountTEO[nvmTE][MaxTEOCopies][nmaxbinsperpage]; // Writing of TE Outer stubs
 
-	if (maskME) {
-		clear2DArray(nvmME, addrCountME);
```

Comment:
can you remind me what these arrays are? It seems like clearing a 2d array or a 3d array would take a long time (many clock cycles). I didn't think we needed to clear these arrays on every new BX.

Reply:
They keep track of the memory address the output stubs should be written to. Because we've got several possible memories to write to, and they have copies, and some of them have bins that also needs to be kept track of, they're multi-dimensional. They are partitioned fully so seems like it can handle it? It's just once every BX. 

If we don't clear it at the start of every BX, the stubs would get written to the wrong address, and not overwrite the stubs from the previous BX. As far as I understand, we still need to clear it every new BX inside the processing module. It was within the memories that we didn't need to clear it?

Before I cleared it before/outside the main loop, but sometimes it didn't clear it properly when looking at the Vivado simulation. Moving it inside the loop seemed to ensure that the arrays got cleared. Don't know if it has anything to do with the rewind option, as the documentation says:

_The code segment before the loop:_
- _Is considered as initialization._
- _Is executed only once in the pipeline._
- _Cannot contain any conditional operations (if-else)._